pkg/compactdictionary/compactdictionary.go

```diff
@@ -41,15 +40,36 @@ func CompactDictionary(mem memory.Allocator, arr *array.Dictionary) (*array.Dict
 
 	newLen := 0
 	keepValues := make([]int, arr.Dictionary().Len())
-	for i := 0; i < indices.Len(); i++ {
+
+	var indices interface{}
+
+	switch arr.Indices().(type) {
```

Comment:
you can do something much simpler here:
```
switch arr := arr.Indices().(type) {
```
then `arr` is already typed to the `case` and you no longer need to type assert anything